presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -852,4 +883,28 @@ private void assertQueryResultCount(String sql, int expectedResultCount)
     {
         assertEquals(getQueryRunner().execute(sql).getRowCount(), expectedResultCount);
     }
+
+    private void assertBigEndian32Conversion(Integer value)
+    {
+        // We use Integer class to allow passing null values.
+        String intLiteral = "" + value;
+        if (value != null && value == Integer.MIN_VALUE) {
+            // Negative literal value is interpreted as unary + absolute value which exceeds allowed integer range.
```

Comment:
@isadikov : Just for my knowledge, please can you run-through an example for this.

Reply:
If I am not mistaken, generally parsing of negative values is done like this `-123` => `UnaryMinus(Literal(123))` which causes an issue for very large numbers and seems to be how Presto does things. 

For Integer.MIN_VALUE, the absolute value is larger than the signed integer that is why we cannot cannot specify the literal as is and need to use the conversion in the test.

Reply:
Got it. Given this code change the refactoring is not worth it. But this is good to know. Thanks !